Description: Add a fallback if thumbnail_limit can't be fetched from gsettings
Author: James Lu <GLolol@overdrivenetworks.com>
Last-Update: 2016-11-12
Bug: https://github.com/gnome-exe-thumbnailer/gnome-exe-thumbnailer/issues/1

 usr/bin/gnome-exe-thumbnailer | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/usr/bin/gnome-exe-thumbnailer b/usr/bin/gnome-exe-thumbnailer
index 74d8cb1..3f382a8 100755
--- a/usr/bin/gnome-exe-thumbnailer
+++ b/usr/bin/gnome-exe-thumbnailer
@@ -185,8 +185,14 @@ case "$THEME" in
 esac
 
 INPUTFILE_SIZE=$(du -b "$INPUTFILE" | cut -f1 -d$'\t')
+# Try to fetch the max. size for thumbnailed executables from gsettings, but fall back to 10485760 bytes if that fails.
 THUMBNAIL_LIMIT=$(gsettings get org.gnome.nautilus.preferences thumbnail-limit | cut -f2 -d' ')
 
+if [[ ! "$THUMBNAIL_LIMIT" ]]
+then
+	THUMBNAIL_LIMIT=10485760
+fi
+
 
 if [[ ${INPUTFILE##*.} = 'msi' ]]
 then

